lib/ukstore/store.c

```diff
@@ -29,3 +29,1194 @@
+#define S32 __do_not_expand__
+#define U32 __do_not_expand__
+#define S64 __do_not_expand__
+#define U64 __do_not_expand__
+#define PTR __do_not_expand__
+
+/**
+ * Case defines used internally for shortening code
+ *
+ * @param entry the entry to do the case for
+ * @param etype the type of the setter
+ * @param ETYPE capital type of the setter
+ * @param var the value to set
+ * @param eparam the setter type
+ */
+
+/* Signed input, unsigned etype */
+#define SETCASE_DOWNCASTSU(entry, etype, ETYPE, var, eparam)		\
+	do {								\
+		case uk_store_get_entry_type(etype):			\
+		if (unlikely(var < 0 ||					\
+			var > (__ ## eparam) __ ## ETYPE ## _MAX))	\
+			return -ERANGE;					\
+		return (entry)->set.etype((__ ## etype) var);		\
+	} while (0)
+
+/* Unsigned input, signed etype */
+#define SETCASE_DOWNCASTUS(entry, etype, ETYPE, var, eparam)	\
+	do {							\
+		case uk_store_get_entry_type(etype):		\
+		if (unlikely(var > __ ## ETYPE ## _MAX))	\
+			return -ERANGE;				\
+		return (entry)->set.etype((__ ## etype) var);	\
+	} while (0)
+
+/* Both signed */
+#define SETCASE_DOWNCASTSS(entry, etype, ETYPE, var, eparam)	\
+	do {							\
+		case uk_store_get_entry_type(etype):		\
+		if (unlikely(var < __ ## ETYPE ## _MIN ||	\
+				var > __ ## ETYPE ## _MAX))	\
+			return -ERANGE;				\
+		return (entry)->set.etype((__ ## etype) var);	\
+	} while (0)
+
+/* Both unsigned */
+#define SETCASE_DOWNCASTUU(entry, etype, ETYPE, var, eparam)	\
+	do {							\
+		case uk_store_get_entry_type(etype):		\
+		if (unlikely(var > __ ## ETYPE ## _MAX))	\
+			return -ERANGE;				\
+		return (entry)->set.etype((__ ## etype) var);	\
+	} while (0)
+
+/* Signed input, unsigned etype */
+#define SETCASE_UPCASTSU(entry, etype, ETYPE, var, eparam)	\
+	do {							\
+		case uk_store_get_entry_type(etype):		\
+		if (unlikely(var < 0))				\
+			return -ERANGE;				\
+		return (entry)->set.etype((__ ## etype) var);	\
+	} while (0)
+
+/* All other cases */
+#define SETCASE_UPCAST(entry, etype, ETYPE, var, eparam)	\
+	do {							\
+		case uk_store_get_entry_type(etype):		\
+		;						\
+		return (entry)->set.etype((__ ## etype) var);	\
+	} while (0)
+
+/**
+ * All setters below use this description.
+ * Checks the ranges of the values and sets a new value using the save setter.
+ *
+ * @param e the entry to call the setter from
+ * @param val the value to set
+ * @return the return value of the setter
+ */
+
+int
+_uk_store_set_u8(struct uk_store_entry *e, __u8 val)
+{
+	UK_ASSERT(e);
+	switch (e->type) {
+	SETCASE_DOWNCASTUS(e, s8, S8, val, u8);
+	SETCASE_UPCAST(e, u8, U8, val, u8);
+	SETCASE_UPCAST(e, u16, U16, val, u8);
+	SETCASE_UPCAST(e, s16, S16, val, u8);
+	SETCASE_UPCAST(e, u32, U32, val, u8);
+	SETCASE_UPCAST(e, s32, S32, val, u8);
+	SETCASE_UPCAST(e, u64, U64, val, u8);
+	SETCASE_UPCAST(e, s64, S64, val, u8);
+	SETCASE_UPCAST(e, uptr, PTR, val, u8);
+
+	case uk_store_get_entry_type(charp): {
+		int ret;
+		char to_set[UK_STORE_NR_MAX_LEN];
+
+		sprintf(to_set, "%hhu", val);
```

Comment:
We have general applicable scan and print formats defined. You should preferably use those since u, d etc are not necessarily the same size across architectures:
snprintf(to_set, `"%"__PRu16`, val)